crates/starknet/src/starknet/mod.rs

```diff
@@ -1184,4 +1187,29 @@ mod tests {
 
         assert_eq!(latest_block.unwrap().block_number(), BlockNumber(2));
     }
+    #[test]
+    fn check_timestamp_of_newly_generated_block() {
+        let config = starknet_config_for_test();
+        let mut starknet = Starknet::new(&config).unwrap();
+
+        Starknet::update_block_context(&mut starknet.block_context);
+        starknet.generate_pending_block().unwrap();
+        starknet.blocks.pending_block.set_timestamp(BlockTimestamp(
+            std::time::SystemTime::now().duration_since(std::time::UNIX_EPOCH).unwrap().as_secs(),
+        ));
+        let pending_block_timestamp = starknet.pending_block().header.timestamp;
+
+        let sleep_duration_secs = 5;
+        thread::sleep(Duration::from_secs(sleep_duration_secs));
+        starknet.generate_new_block(StateDiff::default()).unwrap();
+        let now =
+            std::time::SystemTime::now().duration_since(std::time::UNIX_EPOCH).unwrap().as_secs();
+
+        let block_timestamp = starknet.get_latest_block().unwrap().header.timestamp;
+        // check if the pending_block_timestamp is less than the block_timestamp, by at least the
+        // sleep duration - 1
+        assert!(pending_block_timestamp.0 + sleep_duration_secs - 1 < block_timestamp.0);
```

Comment:
Why not use `<=` in both cases? Also, I think the comments should explain where the `- 1` comes from in these assertions. In fact, I think the clearest way would be to introduce a function like this:
```rust
    /// Uses EPSILON=1
    fn assert_timestamps_almost_equal(t1: u64, t2: u64) {
        assert!(t1.abs_diff(t2) <= 1);
    }
```
and use for both assertions, in the first case it being:
```rust
assert_timestamps_almost_equal(
  pending_block_timestamp.0 + sleep_duration_secs,
  block_timestamp.0,
)
```

Reply:
removed the check with -1 and updated the comment. Didn't introduce the suggested function, because it was used only for 1 check